modules/RTC/TPCUtils.js

```diff
@@ -93,35 +90,49 @@ export class TPCUtils {
         const isScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP;
         let scalabilityMode = this.codecSettings[codec].useKSVC
             ? VideoEncoderScalabilityMode.L3T3_KEY : VideoEncoderScalabilityMode.L3T3;
-        let maxBitrate = codecBitrates.high;
+        const qualityLevel = VIDEO_QUALITY_LEVELS.find(level => level.height === newHeight);
+        let maxBitrate;
 
         if (this._isScreenshareBitrateCapped(localVideoTrack)) {
             scalabilityMode = VideoEncoderScalabilityMode.L1T3;
             maxBitrate = desktopShareBitrate;
         } else if (localVideoTrack.getVideoType() === VideoType.DESKTOP) {
             maxBitrate = codecBitrates.ssHigh;
+        } else if (qualityLevel) {
+            maxBitrate = codecBitrates[qualityLevel.level];
         }
 
         const config = {
             active: newHeight > 0,
             maxBitrate,
             scalabilityMode,
-            scaleResolutionDownBy: HD_SCALE_FACTOR
+            scaleResolutionDownBy: this.l2ScaleFactor
         };
 
         if (newHeight >= height || newHeight === 0 || isScreenshare) {
             return config;
         }
 
-        if (newHeight >= height / SD_SCALE_FACTOR) {
-            config.maxBitrate = codecBitrates.standard;
+        if (!qualityLevel) {
+            return config;
+        }
+
+        config.scaleResolutionDownBy = height / newHeight;
```

Comment:
Do we want to round this in a specific way? Maybe to make it an integer rather than allow fractional scaling?

Reply:
We do need fractional scaling. For example, if we the video is captured at 1080p and we need to scale the stream down to 720p, we need to set the `scaleResolutionDownBy` value to 1.5 and it works as expected. I feel implementing a rounding logic to round the float to one decimal value might introduce unexpected errors. Since the requested/captured resolutions are all constants, we should be safe here. WDYT?